iOSDFULibrary/Classes/Utilities/DFUPackage/Manifest/SoftdeviceBootloaderInfo.swift

```diff
@@ -30,17 +30,53 @@
 
 import Foundation
 
-class SoftdeviceBootloaderInfo: ManifestFirmwareInfo {
-    var blSize: UInt32 = 0
-    var sdSize: UInt32 = 0
+// MARK: - SoftdeviceBootloaderInfo
+
+struct SoftdeviceBootloaderInfo: ManifestFirmware, Codable {
     
-    override init(withDictionary aDictionary : Dictionary<String, AnyObject>) {
-        super.init(withDictionary: aDictionary)
-        if aDictionary.keys.contains("bl_size") {
-            blSize = (aDictionary["bl_size"]!).uint32Value
-        }
-        if aDictionary.keys.contains("sd_size") {
-            sdSize = (aDictionary["sd_size"]!).uint32Value
+    // MARK: Properties
+    
+    let binFile: String?
+    let datFile: String?
+    let secureDFUMetadata: SecureMetadata?
+    let _blSize: UInt32?
+    let _sdSize: UInt32?
+    
+    // MARK: Computed Properties
+    
+    var blSize: UInt32 {
+        return secureDFUMetadata?.blSize ?? _blSize ?? 0
+    }
+    
+    var sdSize: UInt32 {
+        return secureDFUMetadata?.sdSize ?? _sdSize ?? 0
+    }
+    
+    // MARK: CodingKeys
+    
+    enum CodingKeys: String, CodingKey {
+        case binFile = "bin_file"
+        case datFile = "dat_file"
+        case secureDFUMetadata = "info_read_only_metadata"
+        case _blSize = "bl_size"
+        case _sdSize = "sd_size"
+    }
+}
+
+// MARK: - SoftdeviceBootloaderInfo.SecureMetadata
+
+extension SoftdeviceBootloaderInfo {
+
+    struct SecureMetadata: Codable {
```

Comment:
I would rename to `SecureDfuMetadata`, or `ReadOnlyMetadata`, or simply `Metadata` and add a comment, that it is only for informational purpose.